src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -29,13 +29,35 @@ import {
 } from './configure_client_utils';
 
 export const configureClient = async (
-  { dataSourceId, savedObjects, cryptography }: DataSourceClientParams,
+  { dataSourceId, savedObjects, cryptography, testClientDataSourceAttr }: DataSourceClientParams,
   openSearchClientPoolSetup: OpenSearchClientPoolSetup,
   config: DataSourcePluginConfigType,
   logger: Logger
 ): Promise<Client> => {
+  let dataSource;
+  let requireDecryption = true;
+
   try {
-    const dataSource = await getDataSource(dataSourceId!, savedObjects);
+    // configure test client
+    if (testClientDataSourceAttr) {
```

Comment:
Using `testClientDataSourceAttr` as condition seems a bit confuse to me, Wondering if we should pass an explicit flag from the browser side to indicate testing client instead of using the existence of some params. like `isPreFlightClient=true`

Reply:
I am removing all the separate functions to getTestClient, so that we can reuse the same logic for both test client and non-test client. We still need to keep the `testClientDataSourceAttr` even tho we add a flag.

Reply:
Right, I am fine to merge them, and testClientDataSourceAttr is still needed, but in future the api could accept other dataSourceAttr but not necessarily for test client.